packages/animations/test/shared_axis_transition_test.dart

```diff
@@ -0,0 +1,1077 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:animations/src/shared_axis_transition.dart';
+import 'package:flutter/animation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:vector_math/vector_math_64.dart';
+
+void main() {
+  group('SharedAxisTransitionType.horizontal', () {
+    testWidgets(
+      'SharedAxisPageTransitionsBuilder builds a SharedAxisTransition',
+      (WidgetTester tester) async {
+        final AnimationController animation = AnimationController(
+          vsync: const TestVSync(),
+        );
+        final AnimationController secondaryAnimation = AnimationController(
+          vsync: const TestVSync(),
+        );
+
+        await tester.pumpWidget(
+          const SharedAxisPageTransitionsBuilder(
+            transitionType: SharedAxisTransitionType.horizontal,
+          ).buildTransitions<void>(
+            null,
+            null,
+            animation,
+            secondaryAnimation,
+            const Placeholder(),
+          ),
+        );
+
+        expect(find.byType(SharedAxisTransition), findsOneWidget);
+      },
+    );
+
+    testWidgets(
+      'SharedAxisTransition runs forward',
+      (WidgetTester tester) async {
+        final GlobalKey<NavigatorState> navigator = GlobalKey<NavigatorState>();
+        const String bottomRoute = '/';
+        const String topRoute = '/a';
+
+        await tester.pumpWidget(
+          _TestWidget(
+            navigatorKey: navigator,
+            transitionType: SharedAxisTransitionType.horizontal,
+          ),
+        );
+
+        expect(find.text(bottomRoute), findsOneWidget);
+        expect(
+            _getTranslationOffset(
+              bottomRoute,
+              tester,
+              SharedAxisTransitionType.horizontal,
+            ),
+            0.0);
```

Comment:
uber-nit (feel free to ignore): I think that dartfmt formats these multi-line expects nicer if you add a trailing comma.

(here and elsewhere)